scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -8,40 +8,43 @@
 from scipy.sparse.linalg._svdp import _svdp
 from scipy.sparse import csr_matrix, csc_matrix, coo_matrix
 
+
+# dtype_flavour to tolerance
 TOLS = {
     np.float32: 1e-4,
     np.float64: 1e-8,
     np.complex64: 1e-4,
     np.complex128: 1e-8,
 }
 
-_dtype_map = {
-    "single": np.float32,
-    "double": np.float64,
-    "complex8": np.complex64,
-    "complex16": np.complex128,
-}
-
 
 def is_complex_type(dtype):
-    return np.dtype(dtype).char.isupper()
+    return np.dtype(dtype).kind == "c"
 
 
 def is_32bit():
     return sys.maxsize <= 2**32  # (usually 2**31-1 on 32-bit)
 
 
-_dtype_testing = []
-for dtype in _dtype_map:
-    if 'complex' in dtype and is_32bit():
-        # PROPACK has issues w/ complex on 32-bit; see gh-14433
-        marks = [pytest.mark.skip]
-    elif 'complex' in dtype:
-        marks = [pytest.mark.slow]  # type: ignore[list-item]
-    else:
-        marks = []
-    _dtype_testing.append(pytest.param(dtype, marks=marks))
-_dtype_testing = tuple(_dtype_testing)  # type: ignore[assignment]
+def is_windows():
+    return 'win32' in sys.platform
+
+
+_dtypes = []
+for dtype_flavour in TOLS.keys():
+    marks = []
+    if is_complex_type(dtype_flavour):
+        if is_32bit():
+            # PROPACK has issues w/ complex on 32-bit; see gh-14433
+            marks = [pytest.mark.skip]
+        elif is_windows() and np.dtype(dtype_flavour).itemsize == 16:
+            # windows crashes for complex128 (so don't xfail); see gh-15108
+            marks = [pytest.mark.skip]
```

Comment:
I'm tempted to say that we should only skip these if `not has_propack` (c.f. https://github.com/scipy/scipy/pull/16361/files), as `complex128` on windows is not such an extremely exotic datatype that we keep this skip (and accept this rough corner) when propack eventually gets turned on unconditionally.